source/sonata_report.rst

```diff
@@ -86,6 +88,99 @@ For a full compartment reports, you need to specify both ``"sections": "all"`` a
         }
     }
 
+Extension of Compartment report
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To enable fine-grained control over which specific compartments are recorded, SONATA supports an extension using a ``compartment_sets.json`` file. This allows users to define sets of explicit compartment targets (e.g., ``dend[10](0.1)``) for selected node_id.
+
+File: ``compartment_sets.json``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This JSON file defines named sets of compartment targets. Each set specifies the population and a list of ``[node_id, section_id, offset]`` lists:
+
+.. code-block:: json
+
+  {
+    "example_compartment_set": {
+      "population": "S1nonbarrel_neurons",
+      "compartment_set": [
+        [0, 1, 0.1],
+        [3, 5, 0.7],
+        [0, 2, 0.3]
+      ]
+    }
+  }
+
+*   ``population``: The name of the node population of the ``node_id``s in ``compartment_set``.
+*   ``compartment_set``: A list of lists, where each list is ``[node_id, section_id, offset]``. Each of the ``[node_id, section_id, offset]`` lists is referred to as a ``CompartmentLocation``.
+*   ``node_id``: The ID of the node within the specified ``population``.
+*   ``section_index``: The global index of a given section within its cell. NOTE: this is NOT the NEURON section index (e.g. 10 for ``dend[10]``). The ``section_index`` is calculated similar to ``get_section_index`` function of the neurodamus repository in `neurodamus.reports.py <https://github.com/openbraininstitute/neurodamus/blob/1e8b00e55bcc08e9047d6c9a48d068c463c53aef/neurodamus/report.py#L6>`_.
+*   ``offset``: The fractional distance along the section (0<= offset <=1). NOTE: offset for a section is similar to the NEURON segment location e.g. 0.5 in ``dend[10](0.5)``.
+
+
+The order of ``node_id`` in ``compartment_set`` is not necessary. Such entries are acceptable:
```

Comment:
this is ambyguous. What does it mean "not necessary"? Is it sorted automatically or is it kept as is? 
```suggestion
The code preserves the order of ``node_id`` in ``compartment_set``. Such entries are acceptable:
```